rust/routee-compass-core/src/model/cost/vehicle/vehicle_cost_rate.rs

```diff
@@ -0,0 +1,50 @@
+use crate::model::{traversal::state::state_variable::StateVar, unit::as_f64::AsF64, unit::Cost};
+use serde::{Deserialize, Serialize};
+/// a mapping for how to transform vehicle state values into a Cost.
+/// mappings can be a single instance of Raw, Factor, or Offset mapping.
+///
+/// when multiple mappings are specified they are applied sequentially (in user-defined order)
+/// to the state value.
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum VehicleCostRate {
```

Comment:
I think the word "rate" is a little confusing to me here since Raw and Offset are not actually a rate. I wonder if we could find a more general work that captures the idea that it transforms the cost in some way (maybe `VehicleCostTransform` or `VehicleCostOperation`?)

Reply:
i struggled with this name, and like the idea of fixing it, but i'm gonna push this out for future work because the naming convention here also impacts network costs, the cost model, and the query-time argument names.

Reply:
yeah that makes sense to me